hw/xbox/nv2a/nv2a_pgraph.c

```diff
@@ -299,26 +299,82 @@ static uint64_t fnv_hash(const uint8_t *data, size_t len);
 static uint64_t fast_hash(const uint8_t *data, size_t len, unsigned int samples);
 
 /* PGRAPH - accelerated 2d/3d drawing engine */
+
+static uint32_t pgraph_rdi_read(PGRAPHState *pg,
+                                unsigned int select, unsigned int address)
+{
+    uint32_t r = 0;
+    switch(select) {
+    case RDI_INDEX_VTX_CONSTANTS0:
+    case RDI_INDEX_VTX_CONSTANTS1:
```

Comment:
Are we confident `RDI_INDEX_VTX_CONSTANTS1` is valid? Have we confirmed that `0xCC` select works on real hardware? If they address the same data, why would you use `CONSTANTS1` instead of `CONSTANTS0`?

Reply:
> If they address the same data, why would you use CONSTANTS1 instead of CONSTANTS0?

Described in PR description:

> I've added code for select 0x17 and 0xCC as example, which access vertex program constants. **They mostly seem to access the same data (probably for 2 separate transform-engines) and I doubt we'll have to emulate these banks going out of sync**.

---

> Are we confident RDI_INDEX_VTX_CONSTANTS1 is valid? Have we confirmed that 0xCC select works on real hardware?

I believe I did, but I don't fully remember anymore (this code is based on work from 2017; and I haven't  touched it again since February 2019 - when this PR was created - either).

I checked existing code, and found this:

- [nv2a-re only seems to use 0x17](https://github.com/JayFoxRox/nv2a-re/blob/37a425ceb29f863372037c6f6539d1c45f11bd24/main.c#L159-L161).
- [nv2a-trace (a development branch) seems to use both](https://github.com/JayFoxRox/nv2a-trace/blob/b38f2e55473366380a1040658ace68f4169f1dfd/Trace.py#L249-L250).

I can't seem to find any dumps I've made with nv2a-trace though, I'm still somewhat confident it's almost always the same data.